packages/roosterjs-content-model-plugins/lib/imageEdit/ImageEditPlugin.ts

```diff
@@ -411,35 +412,39 @@ export class ImageEditPlugin implements ImageEditor, EditorPlugin {
             this.shadowSpan
         ) {
             editor.formatContentModel(
-                (model, _) => {
+                (model, context) => {
                     const selectedSegmentsAndParagraphs = getSelectedSegmentsAndParagraphs(
                         model,
                         false
                     );
                     if (!selectedSegmentsAndParagraphs[0]) {
                         return false;
                     }
-                    const segment = selectedSegmentsAndParagraphs[0][0];
-
-                    if (
-                        this.lastSrc &&
-                        this.selectedImage &&
-                        this.imageEditInfo &&
-                        this.clonedImage &&
-                        segment.segmentType == 'Image'
-                    ) {
-                        applyChange(
-                            editor,
-                            this.selectedImage,
-                            segment,
-                            this.imageEditInfo,
-                            this.lastSrc,
-                            this.wasImageResized || this.isCropMode,
-                            this.clonedImage
-                        );
-                        segment.isSelected = shouldSelectImage;
-                        segment.isSelectedAsImageSelection = shouldSelectAsImageSelection;
 
+                    const segment = selectedSegmentsAndParagraphs[0][0];
+                    const paragraph = selectedSegmentsAndParagraphs[0][1];
+
+                    if (paragraph && segment.segmentType == 'Image') {
+                        mutateSegment(paragraph, segment, image => {
```

Comment:
This is nice.

And I think we only need to return true for `formatContentModel` when image is really changed, right?

Reply:
Actually you can move this down into the "if" block below.